Sources/TestsCommon/CommonTestUtils.swift

```diff
@@ -122,6 +122,94 @@ open class MongoSwiftTestCase: XCTestCase {
     }
 }
 
+/// Enumerates the different topology configurations that are used throughout the tests
+public enum TestTopologyConfiguration: Decodable {
+    public init(from decoder: Decoder) throws {
+        let str = try decoder.singleValueContainer().decode(String.self)
+        self.init(from: str)
+    }
+
+    case sharded
+    case replicaSet
+    case single
+    case unknown
+
+    public init(from str: String) {
```

Comment:
The only usage of it is in `testChangeStreamSpec`, and now that `TestTopologyConfiguration` is `Decodable`, we can update the `topology` field of `ChangeStreamTest` to be `[TestTopologyConfiguration]` instead of `[String]` as you suggest, and it should just work. Then this initializer will no longer be required.

Reply:
done